RecoEgamma/EgammaIsolationAlgos/python/photonIsolationPUPPI_miniAOD.py

```diff
@@ -0,0 +1,35 @@
+import FWCore.ParameterSet.Config as cms
+from CommonTools.PileupAlgos.Puppi_cff import puppi
+
+puppi.candName = cms.InputTag('packedPFCandidates')
+puppi.vertexName = cms.InputTag('offlineSlimmedPrimaryVertices')
+
+process.egmPhotonIsolationMiniAODPUPPI = cms.EDProducer( "CITKPFIsolationSumProducerForPUPPI",
+			  srcToIsolate = cms.InputTag("slimmedPhotons"),
+			  srcForIsolationCone = cms.InputTag('packedPFCandidates'),
+                          puppiValueMap = cms.InputTag('puppi'),
+			  isolationConeDefinitions = cms.VPSet(
+			   cms.PSet( isolationAlgo = cms.string('PhotonPFIsolationWithMapBasedVeto'), 
+				      coneSize = cms.double(0.3),
+				      isolateAgainst = cms.string('h+'),
+				      miniAODVertexCodes = cms.vuint32(2,3),
+				      vertexIndex = cms.int32(0),
+				    ),
+			   cms.PSet( isolationAlgo = cms.string('PhotonPFIsolationWithMapBasedVeto'), 
+				      coneSize = cms.double(0.3),
+				      isolateAgainst = cms.string('h0'),
+				      miniAODVertexCodes = cms.vuint32(2,3),
+				      vertexIndex = cms.int32(0),
+				    ),
+			   cms.PSet( isolationAlgo = cms.string('PhotonPFIsolationWithMapBasedVeto'), 
+				      coneSize = cms.double(0.3),
+				      isolateAgainst = cms.string('gamma'),
+				      miniAODVertexCodes = cms.vuint32(2,3),
+				      vertexIndex = cms.int32(0),
+				    )
+    )
+  )
+
+	   
+PuppiIsolationPhotonsSequenceMiniAOD = cms.Sequence(puppi + egmPhotonIsolationMiniAODPUPPI)
```

Comment:
Right, packedCandidates have puppiWeight() saved. The puppi module does not need to be rerun.
Instead of reading the transient ValueMap of puppi weights created by rerunning puppi module, it's more appropriate to read this weight from the saved/already computed information.